public/apps/account/tenant-switch-panel.tsx

```diff
@@ -0,0 +1,251 @@
+ *   permissions and limitations under the License.
+ */
+
+import React, { useEffect, useState } from 'react';
+import { CoreStart } from 'kibana/public';
+import {
+  EuiButton,
+  EuiButtonEmpty,
+  EuiCallOut,
+  EuiComboBox,
+  EuiComboBoxOptionOption,
+  EuiModal,
+  EuiModalBody,
+  EuiModalFooter,
+  EuiOverlayMask,
+  EuiRadioGroup,
+  EuiSpacer,
+  EuiText,
+  EuiTitle,
+} from '@elastic/eui';
+import {
+  comboBoxOptionToString,
+  stringToComboBoxOption,
+} from '../configuration/utils/combo-box-utils';
+import { selectTenant } from '../configuration/utils/tenant-utils';
+import { fetchAccountInfo } from './utils';
+
+interface TenantSwitchPanelProps {
+  coreStart: CoreStart;
+  username: string;
+  handleClose: () => void;
+}
+
+const GLOBAL_TENANT_KEY_NAME = 'global_tenant';
+const GLOBAL_TENANT_RADIO_ID = 'global';
+const PRIVATE_TENANT_RADIO_ID = 'private';
+const CUSTOM_TENANT_RADIO_ID = 'custom';
+
+export function TenantSwitchPanel(props: TenantSwitchPanelProps) {
+  const [tenants, setTenants] = useState<{ [key: string]: boolean }>({});
+  const [errorCallOut, setErrorCallOut] = useState<string>();
+
+  useEffect(() => {
+    const fetchData = async () => {
+      try {
+        const accountInfo = (await fetchAccountInfo(props.coreStart.http)).data;
+        setTenants(accountInfo?.tenants);
+      } catch (e) {
+        // TODO: switch to better error display.
+        console.error(e);
+      }
+    };
+
+    fetchData();
+  }, [props.coreStart.http]);
+
+  // Custom tenant combo box related.
+  const [selectedCustomTenantOptions, setSelectedCustomTenantOptions] = useState<
+    EuiComboBoxOptionOption[]
+  >();
+  const onCustomTenantComboChange = (selectedOptions: EuiComboBoxOptionOption[]) => {
+    setSelectedCustomTenantOptions(selectedOptions);
+    setErrorCallOut('');
+  };
+  const customTenantOptions = _.keys(tenants).filter((tenant) => {
+    return tenant !== GLOBAL_TENANT_KEY_NAME && tenant !== props.username;
+  });
+
+  const isMultiTenancyEnabled = props.config.multitenancy.enabled;
+  const isGlobalEnabled = props.config.multitenancy.tenants.enable_global;
+  const isPrivateEnabled = props.config.multitenancy.tenants.enable_private;
+
+  const shouldDisableGlobal = !isGlobalEnabled || !tenants[GLOBAL_TENANT_KEY_NAME];
+  const getGlobalDisabledInstruction = () => {
+    if (!isGlobalEnabled) {
+      return 'Contact the administrator to enable global tenant';
+    }
+
+    if (!tenants[GLOBAL_TENANT_KEY_NAME]) {
+      return 'Contact the administrator to get access to global tenant';
+    }
+  };
+
+  // The key for private tenant is the user name.
+  const shouldDisablePrivate = !isPrivateEnabled || !tenants[props.username];
+  const getPrivateDisabledInstruction = () => {
+    if (!isPrivateEnabled) {
+      return 'Contact the administrator to enable private tenant';
+    }
+
+    if (!tenants[props.username]) {
+      return 'Contact the administrator to get access to private tenant';
+    }
```

Comment:
Non-block: I think this case should not happen, right?

Reply:
I thought this was how "the user is not granted the access" is controlled.